src/components/data-management/DownloadData.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect } from 'react';
+import _ from 'lodash';
+import {
+  Menu, Tooltip, Dropdown, Button,
+} from 'antd';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { saveAs } from 'file-saver';
+import pipelineStatus from '../../utils/pipelineStatusValues';
+import downloadData from '../../utils/data-management/downloadExperimentData';
+import downloadTypes from '../../utils/data-management/downloadTypes';
+import { exportQCParameters, filterQCParameters } from '../../utils/exportQCParameters';
+import { loadBackendStatus } from '../../redux/actions/backendStatus/index';
+
+const DownloadData = (props) => {
+  const {
+    activeProjectUuid,
+  } = props;
+  const dispatch = useDispatch();
+  const activeProject = useSelector((state) => state.projects[activeProjectUuid]);
+  const experimentSettings = useSelector((state) => state.experimentSettings);
+  const backendStatus = useSelector((state) => state.backendStatus);
+  const samples = useSelector((state) => state.samples);
+  const projects = useSelector((state) => state.projects);
+
+  useEffect(() => {
+    if (activeProject?.experiments?.length) {
+      dispatch(loadBackendStatus(activeProject?.experiments[0]));
+    }
+  }, [activeProject]);
+
+  const pipelineHasRun = (experimentId) => (
+    (backendStatus[experimentId]?.status.pipeline?.status === pipelineStatus.SUCCEEDED)
+    && activeProject?.experiments?.length
```

Comment:
Are the `&& activeProject?.experiments?.length` still necessary? Both this and and `gem2hasRun` is passed experimentId from `activeProject.experiments.length[0]`, so this `activeProject?.experiments?.length` must be > 0?

Reply:
Maybe what you can do is first check if experimentId !== undefined, as it may return undefined from activeProject?.experiments[0]